src/Widgets/calendar/GridDay.vala

```diff
@@ -56,21 +57,30 @@ public class DateTime.Widgets.GridDay : Gtk.EventBox {
 
         label = new Gtk.Label (null);
 
+        event_dots = new Gee.HashMap<string, Gtk.Widget> ();
+
+        model.events_added.connect (add_event_dots);
+        model.events_removed.connect (remove_event_dots);
+
         event_grid = new Gtk.Grid ();
         event_grid.halign = Gtk.Align.CENTER;
         event_grid.height_request = 6;
 
-        var grid = new Gtk.Grid ();
+        grid = new Gtk.Grid ();
         grid.halign = grid.valign = Gtk.Align.CENTER;
         grid.attach (label, 0, 0);
-        grid.attach (event_grid, 0, 1);
+
+        if (event_grid.get_parent () == null) {
```

Comment:
Do we have reason to believe that event_grid already has a parent at construction time? It doesn't look like there's any other place where event_grid is attached or added to another widget

Reply:
It seems so. If I set it to `event_grid.get_parent () != null` then no dots are shown for any events.

Reply:
Right, because when this class is constructed `event_grid` should not yet have a parent and there's nowhere else in this class where `event_grid` is attached or added to the grid. So this code might as well say `if (1 == 1)`

Reply:
I see. I'll fix it right away to not have the `if` block.